### DISCUSSION TITLE
convert pdf pages to jpeg based on pdf crop box

### DISCUSSION THREAD

**Discussion by yamikojo** [dlemstra/Magick.NET#1118] [Help]
Hello, how can I set the below option using Magick.NET?

"By default, ImageMagick sets the page size to the MediaBox. Some PDF files, however, have a CropBox that is smaller than the MediaBox and may include white space, registration or cutting marks outside the CropBox. To force ImageMagick to use the CropBox rather than the MediaBox, use -define (e.g. `-define pdf:use-cropbox=true`)."

This is my version of Magick.NET`<PackageReference Include="Magick.NET-Q16-AnyCPU" Version="8.3.1" />`

**Update:**
I came across `PdfReadDefines` but I get an error on trying to use it:
`The type or namespace name 'PdfReadDefines' could not be found (are you missing a using directive or an assembly reference?)`
I have these two imports already. 

```
using ImageMagick;
using ImageMagick.Defines;
```
This is what I am trying to do: (The other lines do not throw any errors)

```
  using var pdf= new MagickImageCollection(input, new MagickReadSettings {
      Density = new Density(300, 300),
      Defines = new PdfReadDefines() {
          UseCropBox = true,
          UseTrimBox = false
      }
  });
```
Thanks for your help.

**Comment by yamikojo** [comment]
@dlemstra sorry for the direct ping. I am still stuck on this. Please let me know if you have any pointers.

**Reply by dlemstra** [reply]
You will also need to add `using ImageMagick.Formats`,
